### PR TITLE
UIDATIMP-717: Cover <ProfileAssociator> with tests

### PR DESCRIPTION
**Pull request by Javokhir12**
## Purpose

Cover `<ProfileAssociator>` component with tests using RTL + Jest
## Approach

Cover `<ProfileAssociator>` component with tests using RTL + Jest
## Refs

 https://issues.folio.org/browse/UIDATIMP-717
## Screenshots

Screenshot (85)

### CODE REVIEW

File: src/components/ProfileAssociator/tests/ProfileAssociator.test.js [hunk 1]
```diff
@@ -0,0 +1,348 @@
+import React from 'react';
+import { BrowserRouter as Router } from 'react-router-dom';
+import {
+  waitFor,
+  fireEvent, act,
+} from '@testing-library/react';
```
Comment (OleksandrHladchenko1) [inline src/components/ProfileAssociator/tests/ProfileAssociator.test.js:6]:
```suggestion
import {
  waitFor,
  fireEvent,
  act,
} from '@testing-library/react';
```

File: src/components/ProfileAssociator/tests/ProfileAssociator.test.js [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+      });
+
+      const searchInput = getByRole('searchbox', { name: /search job profiles/i });
+
+      fireEvent.change(searchInput, { target: { value: 'test' } });
+
+      expect(getByRole('button', { name: /search/i })).toBeEnabled();
+    });
+
+    it('user can clear it', async () => {
```
Comment (OleksandrHladchenko1) [inline src/components/ProfileAssociator/tests/ProfileAssociator.test.js:189]:
```suggestion
    it('user can clear search box', async () => {
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -25,6 +25,8 @@
 * Cover `<UploadingJobsContextProvider>` component with tests (UIDATIMP-729)
 * Cover `<UploadingJobsDisplay` component with tests (UIDATIMP-730)
 * Adjust UI for the Field protection create/edit screen (UIDATIMP-1043)
+* Cover `<ProfileAssociator>` component with tests (UIDATIMP-717)
+
```
Comment (OleksandrHladchenko1) [inline CHANGELOG.md:29]: Remove this unnecessary line, please.

File: src/components/ProfileAssociator/tests/ProfileAssociator.test.js [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+        hasLoaded
+      />
+    </Router>
+  );
+
+  return renderWithIntl(component, translationsProperties);
+};
+
+describe('<ProfileAssociator>', () => {
+  describe('when there is not associated profiles', () => {
```
Comment (mariia-aloshyna) [inline src/components/ProfileAssociator/tests/ProfileAssociator.test.js:139]:
```suggestion
  describe('when there is no associated profiles', () => {
```

File: src/components/ProfileAssociator/tests/ProfileAssociator.test.js [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+        const confirmButton = await findByText('Confirm');
+
+        fireEvent.click(confirmButton);
+
+        await waitFor(() => expect(queryByText('Confirmation')).not.toBeInTheDocument());
+      });
+    });
+  });
+
+  it('profiles is sorted by "Name" by default', () => {
```
Comment (mariia-aloshyna) [inline src/components/ProfileAssociator/tests/ProfileAssociator.test.js:299]:
```suggestion
  it('profiles are sorted by "Name" column by default', () => {
```

**Review by IvanKryzhanovskyi — APPROVED** [review]

**Review by OleksandrHladchenko1 — APPROVED** [review]

**Pull request merged**